src/core/render/embed.js

```diff
@@ -91,7 +91,10 @@ function walkFetchEmbed({ embedTokens, compile, fetch }, cb) {
   }
 }
 
-export function prerenderEmbed({ compiler, raw = '', fetch }, done) {
+export function prerenderEmbed(
+  { compiler, raw = '', fetch, beforeEmbed },
+  done
+) {
```

Comment:
Another thing is, this hook would depend on `marked`, thus is markdown-compiler-specific. What if we change renderers later?

Maybe it is good for now (performance boost) and if we change markdown renderer later then it would simply be a breaking change.

Arguably even without this hook, it would be a breaking change because output of the strings can change. But with this hook in place, it could be that a different markdown renderer does not have a concept of tokens that can be passed to a hook plugin.

Also, what if we wish to add the ability to specify markdown renderers? beforeEach and afterEach would continue to make sense, but maybe beforeEmbed would not make sense if a different markdown renderer does not pass tokens out.

So the question is: do we want to add this, and be coupled to `marked`? Or not?

Reply:
If we decide we want this, I think the way it is implemented looks good.